examples/rsa-pkcs22/src/rsa-pkcs22.rs

```diff
@@ -0,0 +1,217 @@
+
+#[derive(Debug)]
+pub enum Error {
+    InvalidLength,
+    MessageTooLarge,
+}
+
+pub type PK = (RSAInt, RSAInt);
+pub type SK = (RSAInt, RSAInt);
+pub type KeyPair = (PK, SK);
+pub type ByteSeqResult = Result<ByteSeq, Error>;
+pub type RSAIntResult = Result<RSAInt, Error>;
+
+// Encryption primitive, see section 5.1.1 of https://datatracker.ietf.org/doc/html/rfc8017#section-5.1.1
+pub fn rsaep(pk: PK, m: RSAInt) -> RSAIntResult {
+    let (n, e) = pk;
+    if m > n - RSAInt::ONE() {
+        RSAIntResult::Err(Error::MessageTooLarge)
+    } else {
+        RSAIntResult::Ok(m.pow_mod(e, n))
+    }
+}
+
+// Decryption primitive, see section 5.1.2 of https://datatracker.ietf.org/doc/html/rfc8017#section-5.1.2
+pub fn rsadp(sk: SK, c: RSAInt) -> RSAIntResult {
+    let (n, d) = sk;
+    if c > n - RSAInt::ONE() {
+        RSAIntResult::Err(Error::MessageTooLarge)
+    } else {
+        RSAIntResult::Ok(c.pow_mod(d, n))
+    }
+}
+
+// Signature primitive, see section 5.2.1 of https://datatracker.ietf.org/doc/html/rfc8017#section-5.2.1
+pub fn rsasp1(sk: SK, m: RSAInt) -> RSAIntResult {
+    let (n, d) = sk;
+    if m > n - RSAInt::ONE() {
+        RSAIntResult::Err(Error::MessageTooLarge)
+    } else {
+        RSAIntResult::Ok(m.pow_mod(d, n))
+    }
+}
+
+// Verification primitive, see section 5.2.2 of https://datatracker.ietf.org/doc/html/rfc8017#section-5.2.2
+pub fn rsavp1(pk: PK, s: RSAInt) -> RSAIntResult {
+    let (n, e) = pk;
+    if s > n - RSAInt::ONE() {
+        RSAIntResult::Err(Error::MessageTooLarge)
+    } else {
+        RSAIntResult::Ok(s.pow_mod(e, n))
+    }
+}
+
+// RSA integer to octet string primitive, as described in section 4.1 of https://datatracker.ietf.org/doc/html/rfc8017#section-4.1
+pub fn i2osp(x: RSAInt, x_len: u32) -> ByteSeqResult {
+    if x >= (RSAInt::exp(RSAInt::from_literal(256u128), x_len)) 
+            && x_len != BYTE_SIZE {
+        ByteSeqResult::Err(Error::InvalidLength)
+    } else {
+        ByteSeqResult::Ok(RSAInt::to_byte_seq_be(x)
+            .slice((BYTE_SIZE - x_len) as usize, x_len as usize))
+    }
+}
+
+// Octet string to RSA integer primitive, as described in section 4.2 of https://datatracker.ietf.org/doc/html/rfc8017#section-4.2
+pub fn os2ip(x: &ByteSeq) -> RSAInt {
+    RSAInt::from_byte_seq_be(x)
+}
+
+// Mask generation function from appendix B.2.1, see https://datatracker.ietf.org/doc/html/rfc8017#appendix-B.2.1
+pub fn mgf1(mgf_seed: &ByteSeq, mask_len: usize) -> ByteSeqResult {
+    let mut result = ByteSeqResult::Err(Error::InvalidLength);
+    if mask_len < 2usize^32usize * HLEN {
+        let mut t = ByteSeq::new(0);
+        for i in 0..((mask_len + 32) / 32) {
+            let x = i2osp(RSAInt::from_literal(i as u128), 4u32)?;
+            t = t.concat(&sha256(&mgf_seed.concat(&x)));
+        }
+        result = ByteSeqResult::Ok(t.slice(0, mask_len))
+    }
+    result
+}
+
+#[cfg(test)]
+extern crate quickcheck;
+#[cfg(test)]
+#[macro_use(quickcheck)]
```

Comment:
Importing is the recommend way of getting the macro in scope `use quickcheck_macros::quickcheck;`